### ISSUE TITLE
Gradle task assembleDebug failed with exit code 1

### ISSUE THREAD

**Issue by dheeraj2dj** [flutter/flutter#39724]
`flutter run` failing with latest master

## Steps to Reproduce

1. pull latest master (commit id: `16fcb83fecd305e7c576a9366d31d4ff8acddf28`)
2. run `flutter run -v`

## Logs

```
Gradle task assembleDebug failed with exit code 1

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      _buildGradleProjectV2 (package:flutter_tools/src/android/gradle.dart:765:5)
<asynchronous suspension>
#2      buildGradleProject (package:flutter_tools/src/android/gradle.dart:505:14)
<asynchronous suspension>
#3      _AndroidBuilderImpl.buildApk (package:flutter_tools/src/android/android_builder.dart:101:11)
<asynchronous suspension>
#4      AndroidDevice.startApp (package:flutter_tools/src/android/android_device.dart:485:28)
<asynchronous suspension>
#5      FlutterDevice.runHot (package:flutter_tools/src/resident_runner.dart:378:54)
<asynchronous suspension>
#6      HotRunner.run (package:flutter_tools/src/run_hot.dart:252:39)
<asynchronous suspension>
#7      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:460:37)
<asynchronous suspension>
#8      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:490:18)
#9      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#10     _rootRunUnary (dart:async/zone.dart:1132:38)
#11     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#12     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#13     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#14     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#15     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#16     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
#17     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
#18     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#19     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#20     _rootRunUnary (dart:async/zone.dart:1132:38)
#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#22     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#23     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#24     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#25     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#26     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
#27     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
#28     AndroidDevice.isLocalEmulator (package:flutter_tools/src/android/android_device.dart)
#29     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#30     _rootRunUnary (dart:async/zone.dart:1132:38)
#31     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#32     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#33     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#34     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#35     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#36     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
#37     _rootRun (dart:async/zone.dart:1124:13)
#38     _CustomZone.run (dart:async/zone.dart:1021:19)
#39     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#40     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#41     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#42     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#43     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#44     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```

## Output of running `flutter doctor -v`

```
[✓] Flutter (Channel master, v1.9.8-pre.54, on Mac OS X 10.14.6 18G87, locale en-IN)
    • Flutter version 1.9.8-pre.54 at /Users/dheeraj2dj/flutter
    • Framework revision 16fcb83fec (4 hours ago), 2019-09-02 23:44:41 -0400
    • Engine revision e7f9ef6aa0
    • Dart version 2.5.0 (build 2.5.0-dev.4.0 36985859e4)


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
    • Android SDK at /Users/dheeraj2dj/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.36.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.3.0

[✓] Connected device (1 available)
    • SM J400F • 4200404df28d65c7 • android-arm • Android 9 (API 28)

• No issues found!
```

**Comment by ShadyBoshra2012** [comment]
I have the same issue right now !!

**Comment by ShadyBoshra2012** [comment]
I have found the solution, I updated google-services and all dependencies in app/gradle

gradle change
`classpath 'com.google.gms:google-services:4.3.0'`
to
`classpath 'com.google.gms:google-services:4.3.1'`

and all dependencies which these I have

`
dependencies {
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test:runner:1.0.2'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
    api 'com.google.firebase:firebase-core:17.2.0'
    implementation 'com.crashlytics.sdk.android:crashlytics:2.10.1'
    implementation 'com.google.android.gms:play-services-ads:18.2.0'
    implementation 'com.google.firebase:firebase-storage:19.0.0'
    implementation 'com.google.firebase:firebase-auth:19.0.0'
    implementation 'com.google.firebase:firebase-firestore:21.0.0'
    implementation 'com.google.firebase:firebase-messaging:20.0.0'
    implementation 'com.google.firebase:firebase-ads:18.2.0'
    implementation 'com.google.firebase:firebase-analytics:17.2.0'
}
`

**Comment by LaurentiuSimionescu** [comment]
I have the same issue. However I can run the project if I open only the android package as an Android project, doing this there is no way in debugging the dart code, only native.

**Comment by dheeraj2dj** [comment]
@ShadyBoshra2012 mine was already on `com.google.gms:google-services:4.3.1`

**Comment by ShadyBoshra2012** [comment]
> 
> 
> @ShadyBoshra2012 mine was already on `com.google.gms:google-services:4.3.1`

Did you update dependencies versions too ?

**Comment by sageata** [comment]
> > @ShadyBoshra2012 mine was already on `com.google.gms:google-services:4.3.1`
> 
> Did you update dependencies versions too ?

Is there an easy way to do this? Or have to go through each one individually?

**Comment by ShadyBoshra2012** [comment]
> > > @ShadyBoshra2012 mine was already on `com.google.gms:google-services:4.3.1`
> > 
> > 
> > Did you update dependencies versions too ?
> 
> Is there an easy way to do this? Or have to go through each one individually?

I didn't have so much of them, so I did each one individually I think, it's not a lot of work huh 🤔! It's better to try more than having this error 🤷🏻‍♂️.

**Comment by LaurentiuSimionescu** [comment]
Any updates with this issue ?

**Comment by dheeraj2dj** [comment]
@LaurentiuSimionescu I just switched to `dev` channel

**Comment by blasten** [comment]
I'm unable to reproduce this issue with the following steps on master:
```
flutter create my_app
cd my_app
flutter run
```

This seems to be a conflict between platform dependencies. If you are using plugins, please which versions you are using.

Note for myself: it would be useful to print the list of plugins as well as the dependency graph from Gradle, so folks can add this to the issue.

**Comment by LaurentiuSimionescu** [comment]
```
dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter

  flutter_svg: ^0.14.1
  cupertino_icons: ^0.1.2
  sliding_up_panel: ^0.3.5
  cached_network_image: ^1.1.1
  rxdart: ^0.22.2
  connectivity: ^0.4.4
  shared_preferences: ^0.5.3+4
  objectdb: ^1.0.7+1
  http: ^0.12.0+2
  geolocator: ^5.1.3
  test: ^1.6.3
  flutter_launcher_icons: ^0.7.3
  flutter_webview_plugin: ^0.3.7
  shimmer: ^1.0.0
  uuid: ^2.0.2
  launch_review: ^2.0.0
  package_info: ^0.4.0+6
  url_launcher: ^5.1.2

dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_driver:
    sdk: flutter
  intl_translation: ^0.17.7
```

**Comment by blasten** [comment]
@LaurentiuSimionescu I'm able to build debug if you have set up an AndroidX project. Otherwise, the simplest way is to create a new project with `flutter create --androidx android_x_app` and then copy the files over.

**Comment by LaurentiuSimionescu** [comment]
I was able to make it work. I was from build gradle tools version.

**Comment by LaurentiuSimionescu** [comment]
**Update**

How I made my project run (debug, profile, release):

1. First thing that i changed was build gradle tools version with a newer one.
2. Changed gradle distributionUrl
3. Make sure my project is migrated to AndroidX
4. [Androidx-compatibility](https://flutter.dev/docs/development/packages-and-plugins/androidx-compatibility)
5. Manually updated Android modules to have at least compileSDKVersion 28
6. Checked checked for errors with `./gradlew assembleRelease` `./gradlew installRelease`
